test/lib/ansible_test/_data/sanity/pylint/plugins/deprecated.py

```diff
@@ -125,17 +154,71 @@ class AnsibleDeprecatedChecker(BaseChecker):
 
     def __init__(self, *args, **kwargs):
         self.collection_version = None
-        self.is_collection = False
-        self.version_constructor = LooseVersion
+        self.collection_name = None
         super(AnsibleDeprecatedChecker, self).__init__(*args, **kwargs)
 
     def set_option(self, optname, value, action=None, optdict=None):
         super(AnsibleDeprecatedChecker, self).set_option(optname, value, action, optdict)
         if optname == 'collection-version' and value is not None:
-            self.version_constructor = SemanticVersion
-            self.collection_version = self.version_constructor(self.config.collection_version)
-        if optname == 'is-collection':
-            self.is_collection = self.config.is_collection
+            self.collection_version = SemanticVersion(self.config.collection_version)
+        if optname == 'collection-name' and value is not None:
+            self.collection_name = self.config.collection_name
+
+    def _check_date(self, node, date):
+        if not isinstance(date, str):
+            self.add_message('invalid-tagged-date', node=node, args=(date,))
+            return
+
+        matcher = TAGGED_VERSION_RE.match(date)
+        if not matcher:
+            self.add_message('invalid-tagged-date', node=node, args=(date,))
+            return
+
+        collection = matcher.group(1)
+        date_str = matcher.group(2)
+
+        if collection != (self.collection_name or 'ansible.builtin'):
```

Comment:
`collection_name` is either `None` or a "real" collection name, i.e. a truthy value. Or do I understand your comment wrong?

(In case someone passes `--collection-name ''` when manually running `pylint`, it will be treated the same as `None`. But I'm not sure why we would want to support this.)

Reply:
Ah, I misread this.